src/main/java/com/zufar/icedlatte/ai/api/AiChecker.java

```diff
@@ -0,0 +1,79 @@
+package com.zufar.icedlatte.ai.api;
+
+import com.zufar.icedlatte.ai.dto.AiDto;
+import com.zufar.icedlatte.ai.dto.AiResult;
+import com.zufar.icedlatte.ai.dto.TextModerationResult;
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import com.google.gson.Gson;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+// https://platform.openai.com/docs/api-reference/moderations
+public class AiChecker implements AiApi {
+
+    @Value("${ai.url}")
+    protected String url;
+    @Value("${ai.api_key}")
+    protected String apiKey;
+    @Value("${ai.request_body}")
+    protected String requestBody;
+    @Value("${ai.message_replacement_pattern}")
+    protected String messageReplacementPattern;
+
+    @Override
+    public AiDto checkMessage(String message) {
+        try {
+            var request = requestBody.replace(messageReplacementPattern, message);
+            HttpURLConnection con = getHttpURLConnection(url, apiKey, request);
+            int responseCode = con.getResponseCode();
+
+            log.debug("Response Code: " + responseCode);
+
+            BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+            String inputLine;
+            StringBuilder response = new StringBuilder();
+            while ((inputLine = in.readLine()) != null) {
+                response.append(inputLine);
+            }
+            in.close();
```

Comment:
I believe it should be in `finally` section, consider using `try-with-resources`

Reply:
yes, fix it